src/engine/QueryPlanner.cpp

```diff
@@ -1160,10 +1186,16 @@ Variable QueryPlanner::generateUniqueVarName() {
 
 // _____________________________________________________________________________
 QueryPlanner::SubtreePlan QueryPlanner::getTextLeafPlan(
-    const QueryPlanner::TripleGraph::Node& node) const {
+    const QueryPlanner::TripleGraph::Node& node,
+    ad_utility::HashMap<Variable, parsedQuery::TextLimitMetaObject>& textLimits)
+    const {
   AD_CONTRACT_CHECK(node.wordPart_.has_value());
   string word = node.wordPart_.value();
   SubtreePlan plan(_qec);
+  if (!textLimits.contains(node.cvar_.value())) {
+    textLimits[node.cvar_.value()] =
+        parsedQuery::TextLimitMetaObject({}, {}, 0);
+  }
```

Comment:
This is needed because if we for example first run this on a ql:contains-word and after that on a ql:contains-entity statement the TextLimitMetaObject already needs to exists for the ql:contains-word statement so that it can already save the id of the operation to the idsOfMustBeFinishedOperations_ variable.